app/sidekiq/kms_key_rotation/batch_initiator_job.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module KmsKeyRotation
+  class BatchComplete
+    def on_complete(_status, _options)
+      BatchInitiatorJob.perform_async
+    end
+  end
+
+  class BatchInitiatorJob
+    include Sidekiq::Worker
+    include KmsEncryptedModelPatch
+
+    sidekiq_options retry: 5, queue: :low
+
+    RECORDS_PER_BATCH = 10_000
+    RECORDS_PER_JOB = 1_000
+
+    def perform
+      batch = Sidekiq::Batch.new
+      batch.description = "KMS Key Rotation #{batch.bid}"
+      batch.on(:complete, BatchComplete)
+
+      return nil if records.empty?
+
+      batch.jobs do
+        Sidekiq::Client.push_bulk('class' => KmsKeyRotation::RotateKeysJob, 'args' => batched_gids)
+      end
+    end
+
+    def records
+      @records ||= get_records
+    end
+
+    private
+
+    def get_records
+      records = []
+      begin
+        get_models.each do |model|
+          break if records.size == RECORDS_PER_BATCH
+
+          records += model.where.not('encrypted_kms_key LIKE ? ', "v#{kms_version}:%")
```

Comment:
Nit: I probably wouldn't include the whole module just for this one method call. Could do `KmsEncryptedModelPatch.kms_version` instead.

Up to you if you want to change. nbd